integreat_cms/cms/views/translations/translations_management_view.py

```diff
@@ -0,0 +1,95 @@
+import logging
+
+from django.shortcuts import redirect, render
+from django.core.exceptions import PermissionDenied
+from django.utils.decorators import method_decorator
+from django.views.generic import TemplateView
+from django.db import transaction
+
+from ...forms import TranslationsManagementForm
+
+from ...decorators import permission_required
+
+logger = logging.getLogger(__name__)
+
+
+@method_decorator(permission_required("cms.manage_translations"), name="dispatch")
+class TranslationsManagementView(TemplateView):
+    """
+    View for showing the machine translations management options
+    """
+
+    template_name = "translations/translations_management.html"
+
+    def get(self, request, *args, **kwargs):
+        r"""
+        Render translations management interface
+
+        :param request: The current request
+        :type request: ~django.http.HttpRequest
+
+        :param \*args: The supplied arguments
+        :type \*args: list
+
+        :param \**kwargs: The supplied keyword arguments
+        :type \**kwargs: dict
+
+        :return: The rendered template response
+        :rtype: ~django.template.response.TemplateResponse
+        """
+
+        form = TranslationsManagementForm(instance=request.region)
+
+        return render(
+            request,
+            self.template_name,
+            {
+                **self.get_context_data(**kwargs),
+                "current_menu_item": "translations_management",
+                "form": form,
+            },
+        )
+
+    @transaction.atomic
+    def post(self, request, *args, **kwargs):
+        r"""
+        Submit :class:`~integreat_cms.cms.forms.translations.translations_management_form.TranslationsManagementForm` objects.
+
+        :param request: The current request
+        :type request: ~django.http.HttpRequest
+
+        :param \*args: The supplied arguments
+        :type \*args: list
+
+        :param \**kwargs: The supplied keyword arguments
+        :type \**kwargs: dict
+
+        :raises ~django.core.exceptions.PermissionDenied: If user does not have the permission to edit the specific page
+
+        :return: The rendered template response
+        :rtype: ~django.template.response.TemplateResponse
+        """
+
+        if not request.user.has_perm("cms.manage_translations"):
+            raise PermissionDenied(
+                f"{request.user!r} does not have the permission to manage machine translation settings."
+            )
+
+        form = TranslationsManagementForm(
+            data=request.POST,
+            instance=request.region,
+        )
+
+        if not form.is_valid():
+            # Add error messages
+            form.add_error_messages(request)
```

Comment:
Typically, an incorrectly submitted form is directly rendered again instead of redirecting to keep the entered values and allow the template tags like `|add_error_class:"border-red-500"` to work (this needs the invalid form object for rendering, whereas when you redirect to the get view again, the form is initialized again with no data).

Reply:
> (this needs the invalid form object for rendering, whereas when you redirect to the get view again, the form is initialized again with no data)

Is the other method preferred, i.e. should I change the way Iḿ handling the form?

Reply:
For successful form submissions, it's good to do a `redirect` afterwards (e.g. to prevent accidental re-submits when the page is refreshed etc), but when there are errors, I suggest to directly call `render` instead of the redirect to keep all the invalid values in the form and make it easier to correct them.